examples/docker-rootful.yaml

```diff
@@ -44,16 +44,64 @@ provision:
     #!/bin/bash
     set -eux -o pipefail
     command -v docker >/dev/null 2>&1 && exit 0
-    if [ ! -e /etc/systemd/system/docker.socket.d/override.conf ]; then
-      mkdir -p /etc/systemd/system/docker.socket.d
+    readonly override_conf=/etc/systemd/system/docker.socket.d/override.conf
+    if [ ! -e "$override_conf" ]; then
+      mkdir -p $(dirname "$override_conf")
       # Alternatively we could just add the user to the "docker" group, but that requires restarting the user session
-      cat <<-EOF >/etc/systemd/system/docker.socket.d/override.conf
-      [Socket]
-      SocketUser={{.User}}
+      cat <<EOF >"$override_conf"
+    [Socket]
+    SocketUser={{.User}}
     EOF
     fi
     export DEBIAN_FRONTEND=noninteractive
     curl -fsSL https://get.docker.com | sh
+- mode: user  # configure docker under non-root user
```

Comment:
I find the provisioning scripts in `examples/docker-rootful.yaml` and `examples/docker.yaml` to be a bit dense and would like to rework them to be a bit longer, but maybe easier to read/understand/maintain.

I understand they serve as an example on how the `Param` feature is going to be used, but I would still prefer to review them in a separate PR after the feature itself has been merged.

Reply:
Removed changes to `docker{,-rootful}.yaml` from this PR.